### PR TITLE
Added auto language detection of the title

### PR DESCRIPTION
**Pull request by sfi-dannybrady**
Tries to auto-detect the language from the title if you were going to try to use the meta data language but it couldn't find one, there is title text, and you didn't disable this auto-detect feature.

### CODE REVIEW

File: newspaper/article.py [hunk 1]
```diff
@@ -208,8 +209,15 @@ def parse(self):
         self.set_meta_language(meta_lang)
 
         if self.config.use_meta_language:
-            self.extractor.update_language(self.meta_lang)
-            output_formatter.update_language(self.meta_lang)
+            # If there is no meta language provide, but detect_lang_from_title is true, and there is title text, try to detect language
+            if self.meta_lang == '' and self.config.detect_lang_from_title and self.title != '':
```
Comment (owaaa) [inline newspaper/article.py]: maybe be better to just do `not self.meta_lang`. 

```
>>> foo=''
>>> not foo
True
>>> foo='1'
>>> not foo
False
```

Comment (owaaa) [reply newspaper/article.py]: same with title, catch both `''`, and `None`

File: newspaper/configuration.py [hunk 2]
```diff
@@ -51,6 +51,9 @@ def __init__(self):
         # Don't toggle this variable, done internally
         self.use_meta_language = True
 
+        # If use_meta_language is true but cannot be parsed, try detecting lang from title
+        self.detect_lang_from_title = True
```
Comment (owaaa) [inline newspaper/configuration.py]: above comment is interesting, says they toggle it internally. wonder if use case here is slightly different.

File: newspaper/article.py [hunk 3]
```diff
@@ -208,8 +209,15 @@ def parse(self):
...
+                detected_lang = detect(self.title)
```
Comment (owaaa) [inline newspaper/article.py]: hopefully the language codes all line up?

File: newspaper/configuration.py (hunk 2 above)
Comment (sfi-dannybrady) [reply newspaper/configuration.py]: They set `use_meta_language = False` in the `set_language` function, so if the user sets a language don't use the meta language. I was thinking the use case for mine was more like the one below which is 

```
# You may keep the html of just the main article body
self.keep_article_html = False
```

which only gets set if you explicitly change it when setting up your config. I'm basically using `use_meta_language` as a proxy for "figure out the language for me", which is hopefully ok.

File: newspaper/article.py (hunk 3 above)
Comment (sfi-dannybrady) [reply newspaper/article.py]: I don't see anything in newspaper that says where the language codes come from, but langdetect does say it uses ISO 639-1 codes. I've added a check for the detected language to make sure it matches one of the available language codes.

**Pull request merged**